python/ray/dag/compiled_dag_node.py

```diff
@@ -334,34 +361,52 @@ def _get_or_compile(
             elif isinstance(task.dag_node, InputNode):
                 task.output_channel = Channel(
                     buffer_size_bytes=self._buffer_size_bytes,
-                    num_readers=task.num_readers,
+                    num_readers=task.num_readers if task.num_readers else 1,
```

Comment:
When would task.num_readers be None?

Reply:
when the task is an InputNode under multi arg scenario, it's not read by anyone

Reply:
Shouldn't we skip creating the Channel in that case?

Also, it may be clearer if we don't include InputNode as a separate task in the DAG, then? If we allow a task to have multiple outputs (but still one output channel), then we can represent a single InputNode with multiple InputAttributeNodes as one DAG task.

Reply:
right now we actually avoid creating Channel for InputAttributeNodes, given there's always gonna be a single InputNode and we always want a single channel as DAG entry point, a Channel is always created for the InputNode. I'm not very sure about the design tradeoff of seperating InputNode and InputAttributeNode but in the scenario here it's definitely cleaner if we can pack everything into InputNode and not have InputAttributeNode at all